src/Orm/BaseModel/index.ts

```diff
@@ -1886,6 +1886,12 @@ export class BaseModel implements LucidRow {
        * to the relationships
        */
       const relationOptions = cherryPick ? cherryPick[relation.serializeAs] : undefined
+
+      if (typeof relation.serialize === 'function') {
+        result[relation.serializeAs] = relation.serialize(value, key, this, relationOptions)
```

Comment:
If the value is an array, then we must call the `serialize` method for each for the row inside the array. 

Basically, the serialize method should always receive a single value